docs/src/Button.doc.js

```diff
@@ -12,217 +12,390 @@ const card = c => cards.push(c);
+          '-`submit`: Use to submit forms. The button is inside/associated with a form.',
+        ],
+        href: 'type',
+      },
+      {
+        name: 'selected',
+        type: 'boolean',
+        required: false,
+        defaultValue: false,
+        description: [
+          'Control the "selected" state of a button component to toggle binary states.',
+          'Optional with button-role + button-type buttons.',
+        ],
+        href: 'selected',
+      },
+      {
+        name: 'href',
+        type: 'string',
+        required: false,
+        defaultValue: null,
+        description: [
+          'Specify a link URL.',
+          'Required with link-role buttons.',
+        ],
+        href: 'type',
+      },
+      {
+        name: 'ref',
+        type: `React.Ref<'button'> | React.Ref<'a'>`,
+        required: false,
+        defaultValue: null,
+        description:
+          'Forward the ref to the underlying button or anchor element.',
+        href: 'ref',
+      },
+      {
+        name: 'role',
+        type: `'button' | 'link'`,
+        required: false,
+        defaultValue: 'button',
+        description: [
+          `Select a button variant:`,
+          `- 'button': Use to render 'submit' or 'button'-type buttons. The button is rendered as a '<button>'.`,
+          `- 'link': Use for buttons to act like links. The button is rendered as an '<a>'.`,
+          `Required with link-role buttons.`,
+        ],
+        href: 'type',
+      },
+      {
+        name: 'rel',
+        type: `'none' | 'nofollow'`,
+        required: false,
+        defaultValue: 'none',
+        description: 'Optional with link-role buttons.',
+        href: 'type',
+      },
+      {
+        name: 'target',
+        type: `null | 'self' | 'blank'`,
+        required: false,
+        defaultValue: 'null',
+        description: [
+          'Define the frame or window to open the anchor defined on `href`:',
+          '- `null` opens the anchor in the same window.',
+          '- `blank` opens the anchor in a new window.',
+          '- `self` opens an anchor in the same frame.',
+          'Optional with link-role buttons.',
+        ],
+        href: 'type',
       },
     ]}
   />
 );
 
 card(
   <Example
-    name="Example"
+    name="Basic button"
+    id="basic-button"
+    defaultCode={`<Button text="Medium-sized button" inline />`}
+  />
+);
+
+card(
+  <Example
+    name="Types & Roles"
+    id="type"
     defaultCode={`
-<Button text="Medium sized button" inline />
+<Row gap={1}>
+  <Tooltip text="default button: button-role + button-type button">
+    <Button onClick={() => {}} text="Clear search history" inline />
+  </Tooltip>
+  <Tooltip text="form button variant: button-role + submit-type button">
+    <Button type="submit" name='satisfaction-questionaire' text="Submit your response" inline />
+  </Tooltip>
+  <Tooltip text="link button variant: link-role button + button-type">
```

Comment:
Could we simplify the tooltip text? Just show `Link button` / `Submit button` / `Default button` in the tooltip. Also don't think `link-role button + button-type` is what we want

Reply:
DONE